File: src/3-behavioral/6-state/no-state.java
public class ActivityEnrollment {
  private ActivityEnrollmentStateType state;
  public ActivityEnrollment(ActivityEnrollmentStateType state) {
    this.state = state;
  }
  public Object execute(Object data) {
    switch (this.state) {
      case Paying:
        return "payment done : " + data;
      case Booking:
        return "booking done : " + data;
      case Cancelling:
        return "cancelling done : " + data;
    }
  }
}

enum ActivityEnrollmentStateType {
  Paying, Booking, Cancelling
}

File: src/3-behavioral/6-state/no-state.ts
// ! ❌ Bad example not using a state

type ActivityEnrollmentStateType = "Paying" | "Booking" | "Cancelling";

export class ActivityEnrollment {
  state: ActivityEnrollmentStateType = "Booking";

  execute(enrollment: unknown): unknown {
    switch (this.state) {
      case "Paying":
        return "payment done : " + enrollment;
      case "Booking":
        return "booking done : " + enrollment;
      case "Cancelling":
        return "cancelling done : " + enrollment;
    }
  }
}
